main/src/cgeo/geocaching/connector/capability/IIgnoreCapability.java

```diff
@@ -0,0 +1,41 @@
+package cgeo.geocaching.connector.capability;
+
+import cgeo.geocaching.connector.IConnector;
+import cgeo.geocaching.models.Geocache;
+
+import androidx.annotation.NonNull;
+
+/**
+ * Connector capability to ignore caches.
+ */
+public interface IIgnoreCapability extends IConnector {
+    /**
+     * Whether the given cache can be ignored. Typically the implementation
+     * would return constant value regardless of the cache.
+     */
+    boolean canIgnoreCache(@NonNull Geocache cache);
+
+    /**
+     * Whether the given cache can be un-ignored. This is mostly for
+     * geocaching.com service where ignored caches cannot be accessed
+     * from c:geo at all, so ignoring is one-way action.
+     * For modern connectors both should be supported
+     */
+    boolean canRemoveFromIgnoreCache(@NonNull Geocache cache);
+
+    /**
+     * Ignore cache
+     *
+     * @return True - success/False - failure
+     */
+    boolean addToIgnorelist(@NonNull Geocache cache);
+
+    /**
+     * Remove the cache from ignored
+     *
+     * @return True - success/False - failure
+     */
+    boolean removeFromIgnorelist(@NonNull Geocache cache);
```

Comment:
>To me it seems like as if removeFromIgnoreList cannot be triggered via the UI at all currently.

Yep, this PR is just refactor to make sure it can be enabled in the future :)

>And how will cgeo check if a cache is currently part of the ignore list? As it doesn't really make sense to offer an unset option if the cache isn't part of the ignore list at all...

I think this will need much deeper refactor, because atm cgeo doesn't care/remember/know about ignore status at all, it relies on the service (like, it expects service will not show/report this cache at all). 

Our use case is when ignored caches are ignored from map search, but if one searches explicitly by geocode, then one can still find and see this cache.

Reply:
OTOH, offering always both options if it is supported would be fine for me as well, to keep things simple. It's currently just the `SuConnector` anyway.

But a possibility to trigger `removeFromIgnoreList` is must have, otherwise the change doesn't make sense ;-)   (...or have I missed something in the code?)